AirpodsPro Battery/AirpodsPro Battery/ViewModel/AirPodsBatteryViewModel.swift

```diff
@@ -142,6 +142,7 @@ class AirPodsBatteryViewModel: BluetoothAirpodsBatteryManagementProtocol {
                
                 completion(true, .connected, deviceType)
             case .failure( _):
+                self?.resetAllDevicesBatteryState()
```

Comment:
First of all thank you for this PR !

I would prefer that we check first the connection state before resetting. Indeed this method is called quite often and therefore a script running issue can happen but it does not mean that the AirPods are not connected.

I would think on something like that: 

`
let connectionState = self?.connectionStatus ?? .disconnected`

`if connectionState == .disconnected {
self?.resetAllDevicesBatteryState()
}`

Reply:
Good point, though this has been working for me - certainly a good idea to double check!

Reply:
@makeSmartio no need to close the PR. just update it :)